calculator/src/main/java/com/market/banica/calculator/service/ProductServiceImpl.java

```diff
@@ -145,6 +147,8 @@ public void deleteProductFromDatabase(String productName) {
 
         productBase.getDatabase().remove(productName);
 
+        auroraClientSideService.announceInterests(productName);
```

Comment:
Why do we subscribe to this item if we are deleting it from the DB?

Reply:
announce interest method has a role to update a product base in order book, where only existing products are populated with new data from the market, so it has two roles in calculator service when passing changes to order book - to update with new product and to remove existing that are deleted, so no longer info is requested for these products

Reply:
using one method for subscribing and unsubscribing to an item can lead to a lot of confusion and to potential bugs, this is not a good architectural approach, please add another method that handles unsubscribing

Reply:
Done